sdk/dotnet/Pulumi/Deployment/Deployment_Run.cs

```diff
@@ -166,6 +166,128 @@ public static Task<ImmutableArray<Resource>> TestAsync<TStack>(IMocks mocks, Tes
             where TStack : Stack, new()
             => TestAsync(mocks, runner => runner.RunAsync<TStack>(), options);
 
+        /// <summary>
+        /// Used inside TestAsync overloads where users have a function that creates resources
+        /// in which case an internal TestStack is used to create the resources.
+        ///
+        /// This function takes the created resources from the TestStack and filters it out of the created resources
```

Comment:
I'm not sure we should filter out the implicitly created stack. The stack resource will be created even for lambda based deployments, test framework should probably reflect that.

Reply:
Ok lets go with that for now, I'm not 100% convinced but this change is still worth having